projects/plugins/jetpack/_inc/client/recommendations/timer/index.jsx

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import React, { useState, useEffect, useCallback, useRef, useMemo } from 'react';
+
+const Timer = ( { className, expiryDate } ) => {
+	const date = useMemo( () => new Date( expiryDate ), [ expiryDate ] );
+	const [ duration, setDuration ] = useState( date - Date.now() );
+	const intervalRef = useRef();
+
+	const tick = useCallback( () => {
+		setDuration( date - Date.now() );
+	}, [ date, setDuration ] );
+
+	const format = useCallback( ts => {
+		const d = ts / 1000;
+		const days = Math.floor( d / ( 3600 * 24 ) );
+		const hours = Math.floor( ( d - days * 3600 * 24 ) / 3600 );
+		const minutes = Math.floor( ( d - days * 3600 * 24 - hours * 3600 ) / 60 );
+		const seconds = Math.floor( d - days * 3600 * 24 - hours * 3600 - minutes * 60 );
+		const parts = [];
+
+		if ( days > 0 ) {
+			// translators: %d is the number of days, d an abbreviation for days (e.g. 7d). Only translate the latter.
+			parts.push( sprintf( __( '%dd', 'jetpack' ), days ) );
+		}
+
+		if ( hours > 0 ) {
+			// translators: %d is the number of hours, h an abbreviation for hours (e.g. 23h). Only translate the latter.
+			parts.push( sprintf( __( '%dh', 'jetpack' ), hours ) );
+		}
+
+		if ( minutes > 0 ) {
```

Comment:
If there are still hours/ days left, but the countdown is on minute 0 of a given hour, the output reads as `10h 20s` - thoughts on showing all time slots even if the value is 0? `10h 0m 20s`? Or maybe intermediate time slots if a higher value is populated?

Reply:
Good call! I think it makes sense to display `0` if higher units have a non null value indeed.

Reply:
<img width="263" alt="Screen Shot 2022-03-24 at 4 57 41 PM" src="https://user-images.githubusercontent.com/1620183/160016465-9727cf4d-442f-4e78-9a68-f4f445eeaaef.png">
<img width="252" alt="Screen Shot 2022-03-24 at 4 59 32 PM" src="https://user-images.githubusercontent.com/1620183/160016468-80c0a78c-b6b7-4313-9a92-0bff4c6be12e.png">